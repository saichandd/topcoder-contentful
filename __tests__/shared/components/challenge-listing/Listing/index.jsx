import React from 'react';
import _ from 'lodash';
import Renderer from 'react-test-renderer/shallow';

const loadMoreDraft = jest.fn();
const loadMorePast = jest.fn();
const selectBucket = jest.fn();
const setFilterState = jest.fn();
const setSort = jest.fn();

const mockDatas = [{
  activeBucket: 'activeBucket',
  auth: {
    tokenV3: 'token',
    user: {
      handle: 'handle',
    },
  },
  challenges: [],
  loadingDraftChallenges: false,
  loadingPastChallenges: false,
  loadMoreDraft,
  loadMorePast,
  selectBucket,
  setFilterState,
  setSort,
  sorts: {},
}, {
  activeBucket: 'PAST',
  auth: {
    tokenV3: 'token',
    user: {
      handle: 'handle',
    },
  },
  challenges: [],
  loadingDraftChallenges: false,
  loadingPastChallenges: false,
  loadMoreDraft,
  loadMorePast,
  selectBucket,
  setFilterState,
  setSort,
  sorts: {},
}, {
  activeBucket: 'UPCOMING',
  auth: {
    tokenV3: 'token',
    user: {
      handle: 'handle',
    },
  },
  challenges: [],
  loadingDraftChallenges: false,
  loadingPastChallenges: false,
  loadMoreDraft,
  loadMorePast,
  selectBucket,
  setFilterState,
  setSort,
  sorts: {},
}, {
  activeBucket: 'activeBucket',
  auth: {
    tokenV3: 'token',
  },
  challenges: [],
  loadingDraftChallenges: false,
  loadingPastChallenges: false,
  loadMoreDraft,
  loadMorePast,
  selectBucket,
  setFilterState,
  setSort,
  sorts: {},
}, {
  activeBucket: 'ONGOING',
  auth: {
    tokenV3: 'token',
  },
  challenges: [],
  loadingDraftChallenges: false,
  loadingPastChallenges: false,
  loadMoreDraft,
  loadMorePast,
  selectBucket,
  setFilterState,
  setSort,
  sorts: {},
}];

jest.mock('utils/challenge-listing/buckets', () => ({
  getBuckets: () => ({
    activeBucket: {},
    saved: {},
    OPEN_FOR_REGISTRATION: {},
    PAST: {},
    ONGOING: {},
    MY: {},
    UPCOMING: {},
  }),
  BUCKETS: {
    ALL: 'activeBucket',
    SAVED_FILTER: 'saved',
    OPEN_FOR_REGISTRATION: 'OPEN_FOR_REGISTRATION',
    PAST: 'PAST',
    ONGOING: 'ONGOING',
    MY: 'MY',
    UPCOMING: 'UPCOMING',
  },
  isReviewOpportunitiesBucket: () => false,
}));

const Listing = require('components/challenge-listing/Listing').default;

afterAll(() => {
  jest.clearAllMocks();
});

test('Matches shallow shapshot', () => {
  const renderer = new Renderer();

  _.forEach(mockDatas, (data) => {
    renderer.render((
      <Listing {...data} />
    ));
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
