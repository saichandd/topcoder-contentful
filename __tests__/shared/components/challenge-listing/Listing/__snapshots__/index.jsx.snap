// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<div
  className="src-shared-components-challenge-listing-Listing-___style__challengeCardContainer___3p0Ge"
>
  <Bucket
    bucket={Object {}}
    bucketId="MY"
    challenges={Array []}
    communityName={null}
    expand={[Function]}
    expandTag={null}
    expanded={false}
    expandedTags={Array []}
    keepPlaceholders={false}
    loadMore={null}
    loading={false}
    newChallengeDetails={false}
    openChallengesInNewTabs={false}
    setFilterState={[MockFunction]}
    setSort={[Function]}
    sort={null}
    userHandle="handle"
  />
  <Bucket
    bucket={Object {}}
    bucketId="OPEN_FOR_REGISTRATION"
    challenges={Array []}
    communityName={null}
    expand={[Function]}
    expandTag={null}
    expanded={false}
    expandedTags={Array []}
    keepPlaceholders={false}
    loadMore={null}
    loading={false}
    newChallengeDetails={false}
    openChallengesInNewTabs={false}
    setFilterState={[MockFunction]}
    setSort={[Function]}
    sort={null}
    userHandle="handle"
  />
  <Bucket
    bucket={Object {}}
    bucketId="ONGOING"
    challenges={Array []}
    communityName={null}
    expand={[Function]}
    expandTag={null}
    expanded={false}
    expandedTags={Array []}
    keepPlaceholders={false}
    loadMore={null}
    loading={false}
    newChallengeDetails={false}
    openChallengesInNewTabs={false}
    setFilterState={[MockFunction]}
    setSort={[Function]}
    sort={null}
    userHandle="handle"
  />
</div>
`;

exports[`Matches shallow shapshot 2`] = `
<div
  className="src-shared-components-challenge-listing-Listing-___style__challengeCardContainer___3p0Ge"
>
  <Bucket
    bucket={Object {}}
    bucketId="PAST"
    challenges={Array []}
    communityName={null}
    expand={[Function]}
    expandTag={null}
    expanded={true}
    expandedTags={Array []}
    keepPlaceholders={false}
    loadMore={[MockFunction]}
    loading={false}
    newChallengeDetails={false}
    openChallengesInNewTabs={false}
    setFilterState={[MockFunction]}
    setSort={[Function]}
    sort={null}
    userHandle="handle"
  />
</div>
`;

exports[`Matches shallow shapshot 3`] = `
<div
  className="src-shared-components-challenge-listing-Listing-___style__challengeCardContainer___3p0Ge"
>
  <Bucket
    bucket={Object {}}
    bucketId="UPCOMING"
    challenges={Array []}
    communityName={null}
    expand={[Function]}
    expandTag={null}
    expanded={true}
    expandedTags={Array []}
    keepPlaceholders={false}
    loadMore={[MockFunction]}
    loading={false}
    newChallengeDetails={false}
    openChallengesInNewTabs={false}
    setFilterState={[MockFunction]}
    setSort={[Function]}
    sort={null}
    userHandle="handle"
  />
</div>
`;

exports[`Matches shallow shapshot 4`] = `
<div
  className="src-shared-components-challenge-listing-Listing-___style__challengeCardContainer___3p0Ge"
>
  <Bucket
    bucket={Object {}}
    bucketId="OPEN_FOR_REGISTRATION"
    challenges={Array []}
    communityName={null}
    expand={[Function]}
    expandTag={null}
    expanded={false}
    expandedTags={Array []}
    keepPlaceholders={false}
    loadMore={null}
    loading={false}
    newChallengeDetails={false}
    openChallengesInNewTabs={false}
    setFilterState={[MockFunction]}
    setSort={[Function]}
    sort={null}
    userHandle=""
  />
  <Bucket
    bucket={Object {}}
    bucketId="ONGOING"
    challenges={Array []}
    communityName={null}
    expand={[Function]}
    expandTag={null}
    expanded={false}
    expandedTags={Array []}
    keepPlaceholders={false}
    loadMore={null}
    loading={false}
    newChallengeDetails={false}
    openChallengesInNewTabs={false}
    setFilterState={[MockFunction]}
    setSort={[Function]}
    sort={null}
    userHandle=""
  />
</div>
`;

exports[`Matches shallow shapshot 5`] = `
<div
  className="src-shared-components-challenge-listing-Listing-___style__challengeCardContainer___3p0Ge"
>
  <Bucket
    bucket={Object {}}
    bucketId="ONGOING"
    challenges={Array []}
    communityName={null}
    expand={[Function]}
    expandTag={null}
    expanded={true}
    expandedTags={Array []}
    keepPlaceholders={false}
    loadMore={null}
    loading={false}
    newChallengeDetails={false}
    openChallengesInNewTabs={false}
    setFilterState={[MockFunction]}
    setSort={[Function]}
    sort={null}
    userHandle=""
  />
</div>
`;
